scripts/promote-catalog.py

```diff
@@ -0,0 +1,486 @@
+            to_commit.append(os.path.join(relative_catalog_dir, file))
+
+    # copy template file if they don't exist
+    relative_template_file = os.path.join(
+        "operators", operator_name, "catalog-templates", f"v{target_version}.yaml"
+    )
+    source_template = os.path.join(
+        repo_dir, f"operators/{operator_name}/catalog-templates/v{source_version}.yaml"
+    )
+    target_template = os.path.join(repo_dir, relative_template_file)
+    if not os.path.exists(target_template):
+        LOG.info("Copying template %s to %s", source_template, target_template)
+        shutil.copyfile(source_template, target_template)
+        to_commit.append(relative_template_file)
+
+    return to_commit
+
+
+def update_makefile(
+    operator_name: str,
+    repo_dir: str,
+    target_version: str,
+) -> str:
+    """
+    Add target version to the `OCP_VERSIONS` in the Makefile if missing
+
+    Args:
+        operator_name (str): Operator name
+        repo_dir (str): Path to the processed local repository
+        target_version (str): Target version to add
+    """
+    to_commit = ""
+
+    relative_makefile_path = os.path.join("operators", operator_name, "Makefile")
+    makefile_path = os.path.join(repo_dir, relative_makefile_path)
+    with open(makefile_path, "r") as makefile:
+        makefile_content = makefile.readlines()
+
+    makefile_out = []
+    for line in makefile_content:
+        if line.startswith("OCP_VERSIONS"):
+            if target_version not in line:
+                # assuming line looks like:
+                # `OCP_VERSIONS=$(shell echo "v4.12 v4.13 v4.14 v4.15" )`
+                split_point = line.rindex('"')
+                updated = f"{line[:split_point]} v{target_version}{line[split_point:]}"
+                makefile_out.append(updated)
+                to_commit = relative_makefile_path
+            else:
+                makefile_out.append(line)
+        else:
+            makefile_out.append(line)
+
+    with open(makefile_path, "w") as makefile:
+        makefile.writelines(makefile_out)
+
+    return to_commit
+
+
+def triage_operators(
+    operators: list[str], repository: OperatorRepo
+) -> tuple[list[str], list[str], list[str]]:
+    """
+    Triage operators based on promotion strategy
+
+    Args:
+        operators (list[str]): List of operators
+    Returns:
+        tuple ([list[str], ...): Operators triaged
+        into 'never', 'always' and 'review'
+    """
+    never, always, review = [], [], []
+    for operator in operators:
+        config = repository.operator(operator).config
+        strategy = config.get("fbc", {}).get("version_promotion_strategy")
+        if strategy == "never":
+            never.append(operator)
+        elif strategy == "always":
+            always.append(operator)
+        elif strategy == "review-needed":
+            review.append(operator)
+        else:
+            LOG.warning("Unknown promotion strategy for operator %s", operator)
```

Comment:
Just to ease the troubleshooting a bit:

```suggestion
            LOG.warning("Unknown promotion strategy for operator %s: %s", operator, strategy)
```

Reply:
Updated